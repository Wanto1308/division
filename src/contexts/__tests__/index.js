import React, { useState } from 'react';
import { useSelector } from 'react-redux';
import { useLocation } from 'react-router-dom';
import ShallowRenderer from 'react-test-renderer/shallow';
import Context from '../index';

useSelector.mockImplementation(fn => {
  fn({});
  return {
    fetchNotificationCount: jest.fn()
  };
});

const add = jest.fn();
const remove = jest.fn();
const scrollTo = jest.fn();
document.getElementsByTagName = () => [{
  classList: { add, remove },
  scrollTo,
}];

describe('src/contexts', () => {
  test('render', () => {
    useLocation.mockImplementationOnce(() => ({ pathname: '/' }));

    useState
      .mockImplementationOnce(v => [v, jest.fn()])
      .mockImplementationOnce(v => [v()(), jest.fn()]);
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<Context />);
    expect(tree).toMatchSnapshot();
  });

  test('close modal and add modal class', () => {
    const setModal = jest.fn();
    const onCloseModal = jest.fn();
    const setOnCloseModal = jest.fn();
    useState
      .mockImplementationOnce(() => ['modal', setModal])
      .mockImplementationOnce(() => [onCloseModal, setOnCloseModal]);
    const ev1 = { currentTarget: 'tes', target: 'tes' };
    const ev2 = { currentTarget: '', target: 'tes' };
    const result1 = Context(Context.defaultProps);

    result1.props.children[1].props.onClick(ev1);
    expect(onCloseModal).toHaveBeenCalled();

    result1.props.children[1].props.onClick(ev2);
    expect(add).toHaveBeenCalledWith('modal-open');
  });
});
