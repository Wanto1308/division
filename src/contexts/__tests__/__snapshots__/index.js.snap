// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/contexts render 1`] = `
<Provider
  value={
    Object {
      "setModal": [MockFunction] {
        "calls": Array [
          Array [
            null,
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      },
      "setOnCloseModal": [MockFunction],
    }
  }
>
  <div
    className="modal"
    onClick={[Function]}
  />
</Provider>
`;
